hooks/diffy/diffy_trigger_compare_job.php

```diff
@@ -1,65 +1,115 @@
 <?php
+
 /**
- * Script to trigger compare jobs on Diffy API.
- * @see https://diffy.website for more details.
- **/
-list($script, $site, $target_env, $source_branch, $deployed_tag, ) = $argv;
+ * @file
+ * Trigger compare jobs using Diffy API.
+ *
+ * @see https://diffy.website
+ */
+
+list($script, $site, $target_env, $source_branch, $deployed_tag,) = $argv;
+
+// Provide Diffy API key and project id as environment variables or
+// override here.
+define('DIFFY_API_KEY', getenv('DIFFY_API_KEY') ? getenv('DIFFY_API_KEY') : 'DIFFY_API_KEY_PLACEHOLDER');
+define('DIFFY_PROJECT_ID', getenv('DIFFY_PROJECT_ID') ? getenv('DIFFY_PROJECT_ID') : 'DIFFY_PROJECT_ID_PLACEHOLDER');
 
-// Settings. Please provide your own here.
-$api_key = 'AAA';
-$project_id = 111;
+// Default base environment to compare to. Defaults to 'prod'.
+define('DIFFY_BASE_ENVIRONMENT', getenv('DIFFY_BASE_ENVIRONMENT') ? getenv('DIFFY_BASE_ENVIRONMENT') : 'prod');
+
+if (empty(DIFFY_API_KEY)) {
+  print 'Please provide DIFFY_API_KEY';
+  exit(1);
+}
+
+if (empty(DIFFY_PROJECT_ID)) {
+  print 'Please provide DIFFY_PROJECT_ID';
+  exit(1);
+}
+
+if (empty(DIFFY_BASE_ENVIRONMENT)) {
+  print 'Please provide DIFFY_BASE_ENVIRONMENT';
+  exit(1);
+}
 
 // Choose what environments to compare.
 switch ($target_env) {
   case 'dev':
-    $operation = 'prod-dev';
+    $environments = DIFFY_BASE_ENVIRONMENT . '-dev';
     break;
+
   case 'test':
-    $operation = 'prod-stage';
+    $environments = DIFFY_BASE_ENVIRONMENT . '-stage';
     break;
+
   default:
-    echo 'Neither Dev nor Test environment. Do not trigger any job.';
-    exit;
+    exit('Neither Dev nor Test environment. Do not trigger any jobs.');
 }
 
-// Get access token from key.
+print "Getting access token from the key.";
```

Comment:
comments are now actually visible progress messages - something you always want to see in the DevOps pipeline.